### PR TITLE
openrpc.json: Detect drift in openrpc spec

### PR DESCRIPTION
**Pull request by ramin**
## Overview

closes: #2672 

Adds a simple ci check to generate openrpc.json file, and fail if it believes there are updates that are not present in the current openrpc.json file.

comments:

- should we simply add to `go-ci.yml`? I decided not to as it felt like different work that deserved it's own workflow
- possible we could add the `git diff` verify commands to the Makefile, yet this all felt ci specific and not worth polluting Makefile for, leaving the yml ugly vs the Makefile bloated.
- the target branch is hella out of date (go 1.19) we should think about getting main rebased in pretty soon @vgonkivs @distractedm1nd 

## Checklist

- [n/a] New and updated code has appropriate documentation
- [n/a] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

### CODE REVIEW

**Review by renaynay — APPROVED** [review]
Nice one @ramin !

**Comment by renaynay** [comment]
I would defo add this to go-ci.yaml btw @ramin, I think we should run it on every PR - it's not a big check.

File: openrpc.json [hunk 1]
```diff
@@ -1,8569 +1,7685 @@
+--> Generating OpenRPC spec
```
Comment (distractedm1nd) [inline openrpc.json:1]: This will cause problems, can you update the Makefile to remove this echo?

**Review by distractedm1nd — CHANGES_REQUESTED** [review]
Excellent, thank you for this

We can delete the old branch since we will have the new file in main

File: openrpc.json (hunk 1 above)
Comment (ramin) [reply openrpc.json:1]: removed

**Pull request merged**